core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/TimeData.kt

```diff
@@ -0,0 +1,113 @@
+     * much more delay we can add to the last departure without causing any conflict. The delay
+     * would be added to the departure time of the last stop, or to the global departure time if no
+     * stop has been reached yet.
+     */
+    val maxDepartureDelayingWithoutConflict: Double,
+
+    /**
+     * By how much we have added delay to any departure, both train departure time and lengthening
+     * stop duration.
+     */
+    val totalDepartureDelay: Double,
+
+    /**
+     * Time of the next conflict on the given location. Used both to identify edges that go through
+     * the same "opening", and to figure out how much delay we can add locally (with margins).
+     */
+    val timeOfNextConflictAtLocation: Double,
+
+    /**
+     * Time the train has spent moving since its departure. Does not include stop times. Does not
+     * account for engineering allowances that would be added further down the path.
+     */
+    val totalRunningTime: Double,
+
+    /**
+     * Time the train has spent stopped since its departure. Includes both used-requested minimum
+     * stop duration, and extra stop times added to avoid conflicts.
+     *
+     * TODO: to implement variable stop times, this will need to be changed to a list with the
+     *   duration of each stop.
+     */
+    val totalStopTime: Double,
+
+    /**
+     * Global delay that have been added to avoid conflicts on the given element, by delaying the
+     * last departure. This is the value that is added on this specific node/edge.
+     */
+    val delayAddedToLastDeparture: Double = 0.0,
+) {
+    // Getters for different handy values. They may not all be used,
+    // but they're here if we need them. It also shows how the values
+    // are related to each other.
+
+    /** This is the time at which we may start the train path, if there's no extra delay */
+    val earliestDepartureTime = earliestReachableTime - totalRunningTime - totalStopTime
+
+    /**
+     * Time elapsed since the departure time. This may be changed further down the path by
+     * lengthening stop durations or adding engineering allowances.
+     */
+    val timeSinceDeparture = totalRunningTime + totalStopTime
+
+    /** Returns a copy of the current instance, with added travel / stop time. */
+    fun withAddedTime(
+        extraTravelTime: Double,
+        extraStopTime: Double,
+    ): TimeData {
+        return copy(
+            earliestReachableTime = earliestReachableTime + extraTravelTime + extraStopTime,
+            totalRunningTime = totalRunningTime + extraTravelTime,
+            totalStopTime = totalStopTime + extraStopTime
+        )
+    }
+
+    /**
+     * Return a copy of the current instance, with "shifted" time values. Used to create new edges.
+     *
+     * @param timeShift by how much we delay the new element compared to the earliest possible time.
+     *   A value >0 means that we want to arrive later (to avoid a conflict)
```

Comment:
This is completely nitpicky, I don't really care.
```suggestion
     *   A value > 0 means that we want to arrive later (to avoid a conflict)
```